server/src/services/metadata.service.ts

```diff
@@ -478,6 +488,146 @@ export class MetadataService implements OnEvents {
+            x2: region.Area.X + region.Area.W / 2,
+            y2: region.Area.Y + region.Area.H / 2,
+          },
+          region.Area.Unit,
+        ),
+      };
+      faceSet.add(faceResult);
+      nameSet.add(faceResultName);
+    }
+    this.logger.debug(`${faceSet.size} faces detected in ${asset.originalPath}`);
+
+    if (nameSet.size === 0) {
+      return true;
+    }
+
+    // Find all person entries matching the face names we have
+    const matches = await this.personRepository.getManyByName(asset.ownerId, [...nameSet], { withHidden: true });
+
+    // Identify which names are not in our inventory yet
+    const matchesNames = new Set(matches.map((m) => m.name.toLowerCase()));
+    const missing = new Set([...nameSet].filter((item) => !matchesNames.has(item.toLowerCase())));
+
+    if (missing.size > 0) {
+      this.logger.debug(`Creating missing persons: ${[...missing]}`);
+    }
+
+    const newPersons = await this.personRepository.createMany(
+      [...missing].map((name) => ({ ownerId: asset.ownerId, name: name })),
+    );
+    if (!newPersons) {
+      this.logger.error(`Something went wrong creating persons`);
+    }
+
+    // All persons, initially found matching face names in this asset and newly created ones
+    const persons = [...matches, ...newPersons];
+
+    // Loop to build all required faceAssets elements
+    for (const face of faceSet) {
+      const facePerson = persons.find((fp) => fp.name.toLowerCase() == face.Name);
+      if (facePerson) {
+        discoveredFaces.push({
+          id: this.cryptoRepository.randomUUID(),
+          personId: facePerson.id,
+          assetId: asset.id,
+          imageHeight: face.Region.imageHeight,
+          imageWidth: face.Region.imageWidth,
+          boundingBoxX1: face.Region.x1,
+          boundingBoxY1: face.Region.y1,
+          boundingBoxX2: face.Region.x2,
+          boundingBoxY2: face.Region.y2,
+          sourceType: SourceType.EXIF,
+        });
+      }
+    }
+
+    // Update or create faces
+    const faceIds = await this.personRepository.upsertFaces(asset.id, discoveredFaces, SourceType.EXIF);
+    this.logger.debug(`Created ${faceIds.length} faceIds: ${faceIds}`);
+
+    // Updates faceAssetId of the newly created persons
+    for (const person of newPersons) {
+      if (!person.faceAssetId) {
+        const faceId = discoveredFaces.find((df) => df.personId == person.id)?.id;
+        if (faceId) {
+          this.logger.debug(`Updating person ${person.id} faceAssetId to ${faceId}`);
+          person.faceAssetId = faceId;
+        }
+      }
+    }
+
+    // Updates new Person's faceAssetId
+    await this.personRepository.updateMany(
+      newPersons.map((person) => ({ id: person.id, faceAssetId: person.faceAssetId })),
+    );
+
+    // Update new Person's thumbnail
+    await this.updatePersonsFaceAsset([...newPersons]);
+
+    await this.assetRepository.upsertJobStatus({
+      assetId: asset.id,
+      facesRecognizedAt: new Date(),
```

Comment:
I don't think it should set this. This flag means the asset went through face detection, so reusing it here will make it unclear what the asset has actually gone through. This can be covered by `metadataExtractedAt` (which is already set).